src/Reflection/ReflectionClass.php

```diff
@@ -1112,7 +1112,16 @@ private function addStringableInterface(array $interfaces): array
 
         foreach ($this->node->getMethods() as $methodNode) {
             if ($methodNode->name->toLowerString() === '__tostring') {
-                $interfaces[$stringableClassName] = $this->reflectClassForNamedNode(new Node\Name(Stringable::class));
+                try {
+                    $stringableInterfaceReflection = $this->reflectClassForNamedNode(new Node\Name($stringableClassName));
+
+                    if ($stringableInterfaceReflection->isInternal()) {
+                        $interfaces[$stringableClassName] = $stringableInterfaceReflection;
+                    }
+                } catch (IdentifierNotFound) {
```

Comment:
Indeed, going back to this approach, while inefficient, is better than the static approach :|